src/test/java/com/redhat/pantheon/model/api/SlingResourceAdapterFactoryTest.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.pantheon.model.api;
+
+import com.redhat.pantheon.model.Module;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Calendar;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class SlingResourceAdapterFactoryTest {
+
+    private final SlingContext slingContext = new SlingContext();
+
+    SlingResourceAdapterFactory modelAdapterFactory = new SlingResourceAdapterFactory();
+
+    @Test
+    void getAdapter() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:createdBy", "a-user",
+                        "jcr:created", Calendar.getInstance())
+                .commit();
+
+        // When
+        Module adapter = modelAdapterFactory.getAdapter(slingContext.resourceResolver().getResource("/content/module1"),
+                Module.class);
+
+        // Then
+        assertNotNull(adapter);
+        assertEquals("a-user", adapter.createdBy.get());
+        assertNotNull(adapter.created.get());
+    }
+
+    @Test
+    void getUnrecognizedAdapter() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:createdBy", "a-user",
+                        "jcr:created", Calendar.getInstance())
+                .commit();
+
+        // When
+        String adapter = modelAdapterFactory.getAdapter(slingContext.resourceResolver().getResource("/content/module1"),
+                String.class);
+
+        // Then
+        assertNull(adapter);
+    }
```

Comment:
You are correct. I guess I wanted to validate what happened when sending an unrecognized adapter class. This is not testing any of our code. Happy remove if you prefer.

Reply:
Might as well keep it. It could alert us if Sling's behavior ever changes unexpectedly, or could (potentially, although not guaranteed) alert us if we misconfigure the Factory.